examples/any_unique.cpp

```diff
@@ -25,6 +25,7 @@
 template <typename T>
 using is_type_index = std::is_same<std::type_index, T>;
 
+#if __cpp_rtti
```

Comment:
Two thoughts:
 1. I think it'd be good to either use `__cpp_rtti` everywhere, or use `UNIFEX_NO_RTTI` everywhere; and
 2. I wonder whether it'd be worth switching from `std::type_index` to `unifex::type_index` so you can leave this code in the build.